### PR TITLE
VR-5661 Update Client for new Canary rules schema

### PR DESCRIPTION
**Pull request by nhatsmrt**
I'm pretty sure the tests are all updated accordingly.

### CODE REVIEW

**Comment by nhatsmrt** [comment]
For context, the rules are:
```
{
    "canary": [
        {
            "category": "Latency", 
            "rule_id": 1005, 
            "description": "Maximum average latency", 
            "rule_parameters": [
                {
                    "type": "latency_ms", 
                    "description": "Threshold", 
                    "name": "threshold"
                }
            ], 
            "name": "latency_avg_max"
        }, 
        {
            "category": "Latency", 
            "rule_id": 1006, 
            "description": "Maximum p90 latency", 
            "rule_parameters": [
                {
                    "type": "latency_ms", 
                    "description": "Threshold", 
                    "name": "threshold"
                }
            ], 
            "name": "latency_p90_max"
        }, 
        {
            "category": "Error", 
            "rule_id": 1007, 
            "description": "Maximum request error percentage", 
            "rule_parameters": [
                {
                    "type": "percentage", 
                    "description": "Threshold", 
                    "name": "threshold"
                }
            ], 
            "name": "error_4xx_rate"
        }, 
        {
            "category": "Error", 
            "rule_id": 1008, 
            "description": "Maximum server error percentage", 
            "rule_parameters": [
                {
                    "type": "percentage", 
                    "description": "Threshold", 
                    "name": "threshold"
                }
            ], 
            "name": "error_5xx_rate"
        }
    ]
}
```

File: client/verta/verta/deployment/update/rules.py [hunk 1]
```diff
@@ -32,39 +32,56 @@ def _from_dict(rule_dict):
         rule_name = rule_dict['rule_parameters'][0]['name']
         rule_value = float(rule_dict['rule_parameters'][0]['value'])
 
-        RULE_SUBCLASSES = [AverageLatencyThresholdRule, P90LatencyThresholdRule, ErrorRateRule]
+        RULE_SUBCLASSES = [
+            MaximumAverageLatencyThresholdRule, MaximumP90LatencyThresholdRule, MaximumRequestErrorPercentageThresholdRule, MaximumServerErrorPercentageThresholdRule
+        ]
```
Comment (convoliution) [inline client/verta/verta/deployment/update/rules.py]: That's a lot of characters 😆 
```suggestion
        RULE_SUBCLASSES = [
            MaximumAverageLatencyThresholdRule,
            MaximumP90LatencyThresholdRule,
            MaximumRequestErrorPercentageThresholdRule,
            MaximumServerErrorPercentageThresholdRule,
        ]
```

File: client/verta/tests/test_cli/test_endpoint.py [hunk 2]
```diff
@@ -78,10 +78,10 @@ def test_canary_update_endpoint(self, client, created_endpoints, experiment_run,
...
         experiment_run.log_requirements(['scikit-learn'])
 
-        canary_rule = '{"rule": "latency", "rule_parameters": \
-        [{"name": "latency_avg", "value": "0.8"}]}'
-        canary_rule_2 = '{"rule": "error_rate", "rule_parameters": \
-        [{"name": "error_rate", "value": "0.8"}]}'
+        canary_rule = '{"rule": "latency_avg_max", "rule_parameters": \
+        [{"name": "threshold", "value": "0.8"}]}'
+        canary_rule_2 = '{"rule": "error_4xx_rate", "rule_parameters": \
+        [{"name": "threshold", "value": "0.8"}]}'
```
Comment (convoliution) [inline client/verta/tests/test_cli/test_endpoint.py]: While I'm nitpicking: the line break operator `\` is strongly discouraged in Python (especially to split a string omg)

Here, using
```python
json.dumps({
    ...
})
```
might be a good substitute, even if it takes up more lines

**Review by convoliution — CHANGES_REQUESTED** [review]
Some nitpicking

File: client/verta/tests/test_cli/test_endpoint.py (hunk 2 above)
Comment (nhatsmrt) [reply client/verta/tests/test_cli/test_endpoint.py]: Fixed.

File: client/verta/verta/deployment/update/rules.py (hunk 1 above)
Comment (nhatsmrt) [reply client/verta/verta/deployment/update/rules.py]: Fixed.

**Review by convoliution — APPROVED** [review]
Thanks a bunch!

**Pull request merged**